pkg/multicluster/local/local_cluster_manager.go

```diff
@@ -84,14 +83,8 @@ func (cm *ClusterManager) UpdateBlueprint(cluster string, blueprint *v1alpha1.Bl
 	if localCluster, err := cm.GetLocalClusterName(); err != nil || localCluster != cluster {
 		return errors.New("Unregistered cluster: " + cluster)
 	}
-	resource := &v1alpha1.Blueprint{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      blueprint.Name,
-			Namespace: blueprint.Namespace,
-		},
-	}
-	if _, err := ctrl.CreateOrUpdate(context.Background(), cm.Client, resource, func() error {
-		resource.Spec = blueprint.Spec
+	// We use the identity transformation: f(x) = x
+	if _, err := ctrl.CreateOrUpdate(context.Background(), cm.Client, blueprint, func() error {
```

Comment:
I believe that's not how CreateOrUpdate works. Internally it calls Get that actually modifies your blueprint.Spec to the state in the cluster. The original code was written correctly.

Reply:
You're right, I'll fix